runtime/libpgmath/lib/common/acos/fma3/ssacos.cpp

```diff
@@ -77,7 +77,11 @@ float __fss_acos_fma3(float const a)
         _sq = _mm_setr_ps(0.0f, sq, 0.0f, 0.0f);
         p1 = _mm_fmadd_ps(p, _x2_x, F);
 
+#if defined(__clang__) && defined(TARGET_LINUX_ARM64)
+        __m128 pi_mask = (__m128)((long double)_mm_cmpgt_epi32(ZERO, (__m128i)((long double)_a)));
```

Comment:
Intel claims the type of _mm_cmpgt_epi32 is __m128i.  You should change pi_mask 

__m128i _mm_cmpgt_epi32 (__m128i a, __m128i b)
Synopsis
__m128i _mm_cmpgt_epi32 (__m128i a, __m128i b)
#include <emmintrin.h>
Instruction: pcmpgtd xmm, xmm
CPUID Flags: SSE2
Description
Compare packed 32-bit integers in a and b for greater-than, and store the results in dst.

Reply:
The only places pi_mask is used expect it to be __m128, e.g. line below it's used as first argument to `_mm_and_ps()` call, which is declared as such:
```
__m128 _mm_and_ps (__m128 a, __m128 b)
```
I see no need for deferring this conversion.

Reply:
If pi_mask is declared properly, do you still need the two casts?  Is _mm_cmpgt_epi32 a pre-processor macro with its own casts?

Reply:
Yes, `_mm_cmpgt_epi32` is a pre-processor macro with its own cast to `__m128i`. So in places where `__m128i` value is needed, no casting is done, otherwise, explicit cast is required. As I explained above, in this case a cast to `__m128` is required.